### PR TITLE
Undefined vs null

### PR DESCRIPTION
**Pull request by timsuchanek**

### CODE REVIEW

File: src/packages/client/src/runtime/transformDmmf.ts [hunk 1]
```diff
@@ -350,22 +372,25 @@ function getScalarArgs(
...
   kind: DMMF.FieldKind = 'scalar',
+  isNullable: boolean,
 ): DMMF.SchemaArg {
   return {
     name,
     isRelationFilter: kind === 'object',
-    inputType: type.map(t => ({
+    inputType: type.map((t) => ({
       isList,
       isRequired: false,
```
Comment (Jolg42) [inline src/packages/client/src/runtime/transformDmmf.ts:392]: isRequired === false always here? Not sure I understand why 🤔

Comment (timsuchanek) [reply src/packages/client/src/runtime/transformDmmf.ts:392]: Because this is a generated type for the `where` arguments.

Comment (timsuchanek) [reply src/packages/client/src/runtime/transformDmmf.ts:392]: They're always optional

File: src/packages/client/src/runtime/error-types.ts [hunk 2]
```diff
@@ -90,6 +91,17 @@ export interface MissingArgError {
   atMostOne: boolean
 }
 
+/**
+ * If a user incorrectly provided null where she shouldn't have
+ */
+export interface InvalidNullArgError {
```
Comment (Jolg42) [inline src/packages/client/src/runtime/error-types.ts:97]: We probably want to add it to the denylist for code generation

Comment (timsuchanek) [reply src/packages/client/src/runtime/error-types.ts:97]: Indeed. I'll raise a separate PR for it

**Pull request merged**